### PR TITLE
Promisify

### PR DESCRIPTION
**Pull request by crookedneighbor**
### Summary

Updates the public methods to accept promises or callbacks.

### Checklist

- [x] Added a changelog entry
- [x] Ran unit tests
- [x] Add integration app page and tests that uses promises

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -61,8 +62,8 @@ var VERSION = process.env.npm_package_version;
  * @param {string|number} [options.paypalCredit.amount] The amount of the transaction. Required when using the Checkout flow.
  * @param {string} [options.paypalCredit.currency] The currency code of the amount, such as `USD`. Required when using the Checkout flow.
  * @param {string} [options.paypalCredit.buttonStyle] The style object to apply to the PayPal Credit button. The options [found here](https://developer.paypal.com/docs/integration/direct/express-checkout/integration-jsv4/customize-button/) are available. The `label` property cannot be adjusted.
- * @param {function} callback The second argument, `data`, is the {@link Dropin} instance.
- * @returns {void}
+ * @param {function} [callback] The second argument, `data`, is the {@link Dropin} instance. Returns a promise if no callback is provided.
+ * @returns {void|Promise} Returns a promise if no callback is provided.
```
Comment (lilaconlee) [inline src/index.js]: We should add an example with promises

File: src/dropin.js [hunk 2]
```diff
@@ -320,11 +322,11 @@ Dropin.prototype._disableErroredPaymentMethods = function () {
 /**
  * Requests a payment method object which includes the payment method nonce used by by the [Braintree Server SDKs](https://developers.braintreepayments.com/start/hello-server/). The structure of this payment method object varies by type: a {@link Dropin~cardPaymentMethodPayload|cardPaymentMethodPayload} is returned when the payment method is a card, a {@link Dropin~paypalPaymentMethodPayload|paypalPaymentMethodPayload} is returned when the payment method is a PayPal account. If a payment method is not available, an error will appear in the UI and and error will be returned in the callback.
```
Comment (lilaconlee) [inline src/dropin.js]: We might want to update the "error will be returned in the callback"

File: test/app/promise.html [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+      var prop = decodeURIComponent(pair[0]);
+      var value = decodeURIComponent(pair[1]);
+
+      // this allows us to pass arrays, booleans, and objects in
+      // the query string. If the parsing fails, we fall back to
+      // a string value.
+      try {
+        jsonParsedValue = JSON.parse(value);
+        value = jsonParsedValue;
+
```
Comment (quinnjn) [inline test/app/promise.html]: Extra space

File: test/app/script-tag-integration.html [hunk 4]
```diff
@@ -44,5 +44,6 @@
   ></script>
   <input id="pay-button" type="submit" value="Pay">
 </form>
+
```
Comment (quinnjn) [inline test/app/script-tag-integration.html]: Unsure if this extra space is intended.

**Comment by Deadarius** [comment]
Hello! Any ETA on when a package with this included is going to be published to npm?

**Comment by crookedneighbor** [comment]
version 1.4.0 includes this and is out and ready to be used!

**Pull request merged**